src/test/java/org/jabref/logic/bibtexkeypattern/BibtexKeyGeneratorTest.java

```diff
@@ -943,14 +943,14 @@ void crossrefkeywordsNKeywordsSeparatedBySpace() {
     @Test
     void testCheckLegalKeyEnforceLegal() {
         assertEquals("AAAA", BibtexKeyGenerator.cleanKey("AA AA", true));
-        assertEquals("SPECIALCHARS", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,^", true));
+        assertEquals("SPECIALCHARS", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,", true));
         assertEquals("", BibtexKeyGenerator.cleanKey("\n\t\r", true));
     }
 
     @Test
     void testCheckLegalKeyDoNotEnforceLegal() {
         assertEquals("AAAA", BibtexKeyGenerator.cleanKey("AA AA", false));
-        assertEquals("SPECIALCHARS#~^", BibtexKeyGenerator.cleanKey("SPECIAL CHARS#{\\\"}~,^", false));
```

Comment:
Oh, wow, did not expect that. See my review comment on allowing characters breaking the parsing of a `.bib` file.

Reply:
Following your comment should i add `{` and `}` and `"` to `KEY ILLEGAL CHARACTERS` then test result after cleaning key will be less about characters which can be cause parsing errors?